src/darkleaf/di/core.clj

```diff
@@ -86,7 +87,7 @@
   (let [ctx           (update ctx :under-construction conj key)
         factory       (registry key)
         declared-deps (p/dependencies factory)
-        resolved-deps (resolve-deps ctx declared-deps)
+        resolved-deps (resolve-deps (assoc ctx :*current-key key) declared-deps)
```

Comment:
`*` is a mark for volatile

Reply:
@darkleaf I can remove `*`, I just used already existing key:
 https://github.com/darkleaf/di/blob/497a2497dc171c0556f3f48df355090decfcfbbd/src/darkleaf/di/core.clj#L85C45-L85C57